docs/EARLY-RELEASE.md

```diff
@@ -0,0 +1,64 @@
+# How to determine if an early patch release is warranted
+
+In the curl project we do releases every 8 weeks. Unless we break the cycle
+and do an early patch release.
+
+We do frequent releases partly to always have the next release "not too far
+away".
+
+## Bugfix
+
+During the release cycle, and especially in the beginning of a new cycle,
+there are times when a bug is reported and after it has been subsequently
+fixed correctly, the discussion might be brought up: is this bug and
+associated fix important enough for an early patch release.
+
+The question can only be properly asked once a fix has been created and landed
+in the git master branch.
+
+## Early release
+
+An early patch release means that we ship a new, complete and full release
+called `major.minor.patch` where the `patch` part is increased by one since
+the previous release. A curl release is a curl release. There is no small or
+big and we never release just a patch. There is only "release".
+
+## Questions to ask
+
+ - Is there a security advisory rated high or critical?
```

Comment:
Come to think of it, if there was, wouldn't that warrant a new release which bumps the minor rev and not just the patch rev? This only applies to *high* and *critical* of course.

Reply:
Would it really? If we just fix an internal flaw somewhere that is still just a patch release, even if that fixes a really serious security flaw - I think.

Reply:
If you are on a monthly cycle, and this fix comes out 2 days into the cycle and you want to release it as a patch, but will be more thoroughly packaging the fix by the end of the cycle, perhaps it is a patch, not a minor release.

Reply:
If we are past the opening of the feature window then all releases are by definition minor rev bumps since new features will have gone in. (or well, *may* have gone in..)

Reply:
> Would it really? If we just fix an internal flaw somewhere that is still just a patch release, even if that fixes a really serious security flaw - I think.

For signalling purposes I think it can be argued for minor rev bump, but technically it should be a patch following the current process.